parser/src/Parse/Expression.hs

```diff
@@ -234,8 +234,8 @@ caseExpr elmVersion =
       (e, multilineSubject) <- trackNewline $ (\(pre, e, post) -> Commented pre e post) <$> padded (expr elmVersion)
       reserved elmVersion "of"
       firstPatternComments <- whitespace
-      result <- cases firstPatternComments
-      return $ E.Case (e, multilineToBool multilineSubject) result
+      (branches, multilineFirstBranch) <- cases firstPatternComments
+      return $ E.Case (e, multilineSubject) (branches, multilineFirstBranch)
```

Comment:
Was it in a separate issue you were suggesting using the linebreak of the first item to control multiline for other kinds of expressions besides cases?  I'm still hesitant about that.  I'd be more comfortable having this only be `JoinAll` if *all* of the branches are single line,  and then maybe doing a separate experiment after adding this to try the idea of having the first item control everything.

Reply:
I thought we could experiment it with only `case` expressions, but you're right, it's better to introduce the single `if` and `case` expressions in a consistent way with existing behavior, then discuss eventually the other method. I will make the change.

Reply:
New version uses JoinAll if all branches are single line.